src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -133,17 +136,31 @@ internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, Ca
             }
         }
 
-        private async Task<IEslintBridgeClient> GetJavaScriptClient(CancellationToken token)
+        private async Task<IEslintBridgeClient> GetJavaScriptClient(CancellationToken cancellationToken)
         {
             var port = await eslintBridgeProcess.Start();
 
-            if (port != javascriptServerPort)
+            try
```

Comment:
It's not immediately obvious what this code is trying to achieve i.e. _Ensure init-linter is called successfully once and only once per process_.
I think at least a comment is required, unless it can be made obvious by renaming/restructuring.

Reply:
Hmm, it's not just the init-linter. Thinking out loud: this method ensures that you get an instance of IEslintBridgeClient that is initialized to a (hopefully still) running port of the eslint-bridge process. So the method wraps 2 things: 1) that you have a running process, and 2) that process was initialized with init-linter. 

So perhaps renaming it from `GetJavaScriptClient` to `GetInitializedEslintBridgeClient` ? 
Or, change it to void and name it `EnsureEslintBridgeClientIsInitialized` - I originally returned an instance of client to ensure that you're getting an initialized instance, but I think there is a problem either way (that you spotted in the other comment):
1) Thread A calls EnsureEslintBridgeClientIsInitialized, client instance is initialized to 888
2) Before thread A manages to call analyze-js, the process dies
3) Thread B calls EnsureEslintBridgeClientIsInitialized, client instance is initialized to 999 but init-linter was not called yet.
4) Thread A attempts to call analyze-js and fails since thread B haven't called init-linter yet.

In either case, I think if the process died in the middle of an analysis, that analysis is gonna fail one way or another.

So if it will be a void, there is a potential for a different problem:
1) Thread A calls GetJavaScriptClient, returns client initialized to 888
2) Before thread A manages to call analyze-js, the process dies
3) Thread B calls GetJavaScriptClient, returns client initialized to 999
4) Thread A attempts to call analyze-js and fails since thread B disposed this instance

Reply:
We are going to have a more complicated situation with TypeScript, where we'll need to check which `ts-config` was last used to initialize the eslintbridge. But that's a problem for another day.

Renaming the method would do.

Let's not over-complicate it for now. From what you said, the other SL flavours don't recover and it doesn't seem to have caused major problems so far.